src/validator.test.ts

```diff
@@ -232,6 +232,34 @@ describe('validate', function () {
     });
   });
 
+  describe('attribute validation', () => {
+    it('should return error on failure to match array', () => {
```

Comment:
If you are up for it, it might make sense to move the other attribute validation tests into this `describe` block

Reply:
makes sense, I will do in another PR since it involves factoring out the schema file